### PR TITLE
Newsletter

### PR DESCRIPTION
**Pull request by lurldgbodex**
## Description
create an endpoint that allows user to subscribe to a newsletter. The subscribed user is sent an email upon subscription success.

## Expected Request
- Post request `/api/v1/newsletter-subscription`
```json
 {
    "email": "john@doe.com"
 }
```

## Success Response
```json
{
  "status_code": 201,
  "message": "string"
```

## Acceptance Criteria
- user is subscribed to the newsletter
- email notification is sent to user of the subscription

## Reference
- [newsletter issue](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/890)

### CODE REVIEW

File: src/main/java/hng_java_boilerplate/user/serviceImpl/UserServiceImpl.java [hunk 1]
```diff
@@ -160,6 +162,20 @@ public User findUser(String id) {
         return userRepository.findById(id).orElseThrow(() -> new NotFoundException("User not found"));
     }
 
+    @Override
+    public CustomResponse forgotPassword(ForgotPasswordRequest passReq, HttpServletRequest request) {
```
Comment (Ofektom) [inline src/main/java/hng_java_boilerplate/user/serviceImpl/UserServiceImpl.java]: This is still here even when the PR has been closed

File: src/test/java/hng_java_boilerplate/user/verify_oto_test/UserServiceImplTest.java [hunk 2]
```diff
@@ -128,4 +130,22 @@ public void testVerifyOtpUserNotFound() {
 
         assertEquals("User not found with email: notfound@example.com", exception.getMessage());
     }
+
+
+    @Test
+    public void testForgotPassword() {
```
Comment (Ofektom) [inline src/test/java/hng_java_boilerplate/user/verify_oto_test/UserServiceImplTest.java]: And this as well

**Review by Fifetoyi — APPROVED** [review]
lgtm

**Review by Adewale66 — COMMENTED** [review]
LGTM @Am0du

**Pull request merged**